dbt/adapters/databricks/connections.py

```diff
@@ -232,6 +257,102 @@ def extract_cluster_id(cls, http_path: str) -> Optional[str]:
     def cluster_id(self) -> Optional[str]:
         return self.extract_cluster_id(self.http_path)  # type: ignore[arg-type]
 
+    def authenticate(self, in_provider: CredentialsProvider) -> CredentialsProvider:
+        self.validate_creds()
+        if self._credentials_provider:
+            return self._provider_from_dict()
+        if in_provider:
+            self._credentials_provider = in_provider.as_dict()
+            return in_provider
+
+        # dbt will spin up multiple threads. This has to be sync. So lock here
+        self._lock.acquire()
+        try:
+            if self.token:
```

Comment:
it looks like you're re-inventing the `DefaultCredentials` and skipping the other unified auth mechanisms - https://github.com/databricks/databricks-sdk-py/blob/main/databricks/sdk/core.py#L285-L314

may I ask why?

Reply:
because I specifically do not want the other auth mechanisms. This is about oauth and oauth only.